### ISSUE TITLE
jvm crash when running ./profiler.sh start -d 60 command by using libasyncProfiler-linux-aarch64.so under Alpine Linux aarch64

### ISSUE THREAD

**Issue by tdy218** [async-profiler/async-profiler#741]
hs_err_pid17.log
```log
Stack: [0x0000ffff60bf0000,0x0000ffff60dffac8],  sp=0x0000ffff60dfe520,  free space=2105k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libstdc++.so.6+0x1683dc]
C  [libstdc++.so.6+0xd9b9c]  std::locale::operator=(std::locale const&)+0x54
C  [libstdc++.so.6+0xd9094]  std::ios_base::_M_init()+0x4c
C  [libstdc++.so.6+0x1153c0]  std::basic_ios<char, std::char_traits<char> >::init(std::basic_streambuf<char, std::char_traits<char> >*)+0x18
C  [ArthasJniLibrary5656461273075369895.tmp+0x273ac]  Symbols::parseLibraries(NativeCodeCache**, int volatile&, int, bool)+0xf4
C  [ArthasJniLibrary5656461273075369895.tmp+0x20208]  VM::ready()+0x28
C  [ArthasJniLibrary5656461273075369895.tmp+0x20c48]  VM::init(JavaVM_*, bool)+0x430
C  [ArthasJniLibrary5656461273075369895.tmp+0x210e0]  JNI_OnLoad+0x14
C  [libjava.so+0xecc4]  Java_java_lang_ClassLoader_00024NativeLibrary_load+0xb4
j  java.lang.ClassLoader$NativeLibrary.load(Ljava/lang/String;Z)V+0
j  java.lang.ClassLoader.loadLibrary0(Ljava/lang/Class;Ljava/io/File;)Z+328
j  java.lang.ClassLoader.loadLibrary(Ljava/lang/Class;Ljava/lang/String;Z)V+48
j  java.lang.Runtime.load0(Ljava/lang/Class;Ljava/lang/String;)V+57
j  java.lang.System.load(Ljava/lang/String;)V+7
j  one.profiler.AsyncProfiler.getInstance(Ljava/lang/String;)Lone/profiler/AsyncProfiler;+23
j  com.taobao.arthas.core.command.monitor200.ProfilerCommand.profilerInstance()Lone/profiler/AsyncProfiler;+165
j  com.taobao.arthas.core.command.monitor200.ProfilerCommand.process(Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+43
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl.process(Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+34
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl.access$100(Lcom/taobao/arthas/core/shell/command/impl/AnnotatedCommandImpl;Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+2
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl$ProcessHandler.handle(Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+5
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl$ProcessHandler.handle(Ljava/lang/Object;)V+5
j  com.taobao.arthas.core.shell.system.impl.ProcessImpl$CommandProcessTask.run()V+11
J 4882 C2 java.util.concurrent.Executors$RunnableAdapter.call()Ljava/lang/Object; (14 bytes) @ 0x0000ffff768c1c44 [0x0000ffff768c1c00+0x44]
j  java.util.concurrent.FutureTask.run()V+42
J 5700 C2 java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run()V (59 bytes) @ 0x0000ffff76122e78 [0x0000ffff76122c00+0x278]
j  java.util.concurrent.ThreadPoolExecutor.runWorker(Ljava/util/concurrent/ThreadPoolExecutor$Worker;)V+95
j  java.util.concurrent.ThreadPoolExecutor$Worker.run()V+5
j  java.lang.Thread.run()V+11
v  ~StubRoutines::call_stub
V  [libjvm.so+0x5be134]  JavaCalls::call_helper(JavaValue*, methodHandle*, JavaCallArguments*, Thread*)+0xb14
V  [libjvm.so+0x5bbc14]  JavaCalls::call_virtual(JavaValue*, KlassHandle, Symbol*, Symbol*, JavaCallArguments*, Thread*)+0x25c
V  [libjvm.so+0x5bc174]  JavaCalls::call_virtual(JavaValue*, Handle, KlassHandle, Symbol*, Symbol*, Thread*)+0x64
V  [libjvm.so+0x6561a0]  thread_entry(JavaThread*, Thread*)+0x68
V  [libjvm.so+0x993b18]  JavaThread::thread_main_inner()+0xc8
V  [libjvm.so+0x993d50]  JavaThread::run()+0x200
V  [libjvm.so+0x84b374]  java_start(Thread*)+0x16c
C  [ld-musl-aarch64.so.1+0x5bbc0]
C  [ld-musl-aarch64.so.1+0x5a418]

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  java.lang.ClassLoader$NativeLibrary.load(Ljava/lang/String;Z)V+0
j  java.lang.ClassLoader.loadLibrary0(Ljava/lang/Class;Ljava/io/File;)Z+328
j  java.lang.ClassLoader.loadLibrary(Ljava/lang/Class;Ljava/lang/String;Z)V+48
j  java.lang.Runtime.load0(Ljava/lang/Class;Ljava/lang/String;)V+57
j  java.lang.System.load(Ljava/lang/String;)V+7
j  one.profiler.AsyncProfiler.getInstance(Ljava/lang/String;)Lone/profiler/AsyncProfiler;+23
j  com.taobao.arthas.core.command.monitor200.ProfilerCommand.profilerInstance()Lone/profiler/AsyncProfiler;+165
j  com.taobao.arthas.core.command.monitor200.ProfilerCommand.process(Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+43
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl.process(Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+34
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl.access$100(Lcom/taobao/arthas/core/shell/command/impl/AnnotatedCommandImpl;Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+2
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl$ProcessHandler.handle(Lcom/taobao/arthas/core/shell/command/CommandProcess;)V+5
j  com.taobao.arthas.core.shell.command.impl.AnnotatedCommandImpl$ProcessHandler.handle(Ljava/lang/Object;)V+5
j  com.taobao.arthas.core.shell.system.impl.ProcessImpl$CommandProcessTask.run()V+11
J 4882 C2 java.util.concurrent.Executors$RunnableAdapter.call()Ljava/lang/Object; (14 bytes) @ 0x0000ffff768c1c44 [0x0000ffff768c1c00+0x44]
j  java.util.concurrent.FutureTask.run()V+42
J 5700 C2 java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run()V (59 bytes) @ 0x0000ffff76122e78 [0x0000ffff76122c00+0x278]
j  java.util.concurrent.ThreadPoolExecutor.runWorker(Ljava/util/concurrent/ThreadPoolExecutor$Worker;)V+95
j  java.util.concurrent.ThreadPoolExecutor$Worker.run()V+5
j  java.lang.Thread.run()V+11
v  ~StubRoutines::call_stub
```

Alpine Linux v3.16.5 aarch64 with libstdc+、libc6-compat installing.

```shell
root# apk list 2>&1 | grep -iE "libstdc++|compat"
libstdc++-11.2.1_git20220219-r2 aarch64 {gcc} (GPL-2.0-or-later LGPL-2.1-or-later) [installed]
libc6-compat-1.2.3-r2 aarch64 {musl} (MIT) [installed]
```

**Comment by tdy218** [comment]
But when i manually compilated from source code under Alpine Linux v3.16.5 aarch64, it just works fine.
The following output is under Alpine Linux v3.16.5 aarch64 and without libstdc+、libc6-compat
```shell
[root@microservice-testing-group-1-v6-8558964d5b-xrwx6:async-profiler]# ldd libasyncProfiler-linux-arm64.so 
        /lib/ld-musl-aarch64.so.1 (0xffff82fc0000)
        libc.musl-aarch64.so.1 => /lib/ld-musl-aarch64.so.1 (0xffff82fc0000)
[root@microservice-testing-group-1-v6-8558964d5b-xrwx6:async-profiler]# 
[root@microservice-testing-group-1-v6-8558964d5b-xrwx6:async-profiler]# ldd /home/admin/.opt/ArmsAgent/arthas/async-profiler/libasyncProfiler-linux-aarch64.so 
        /lib/ld-musl-aarch64.so.1 (0xffffab300000)
        libdl.so.2 => /lib/ld-musl-aarch64.so.1 (0xffffab300000)
        libpthread.so.0 => /lib/ld-musl-aarch64.so.1 (0xffffab300000)
Error loading shared library libstdc++.so.6: No such file or directory (needed by /home/admin/.opt/ArmsAgent/arthas/async-profiler/libasyncProfiler-linux-aarch64.so)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /home/admin/.opt/ArmsAgent/arthas/async-profiler/libasyncProfiler-linux-aarch64.so)
        libc.so.6 => /lib/ld-musl-aarch64.so.1 (0xffffab300000)
Error loading shared library ld-linux-aarch64.so.1: No such file or directory (needed by /home/admin/.opt/ArmsAgent/arthas/async-profiler/libasyncProfiler-linux-aarch64.so)
```

**Comment by tdy218** [comment]
Will the libasyncProfiler  aarch64  musl version be output ?

**Comment by apangin** [comment]
The issue will be addressed in #733.
As of now, there is no other solution than building a dedicated binary for the specific target platform.
